src/c_api/c_api.cc

```diff
@@ -248,11 +251,24 @@ void CustomFComputeDispatcher(const std::string op_name,
   // get actual cudaStream_t out of mxnet gpu stream and pass to lib_api.h
   void *cuda_stream = nullptr;
 #if MXNET_USE_CUDA
-  if (inputs[0].ctx().dev_mask() == Context::kGPU) {
+  if (inputs.size() > 0 && inputs[0].ctx().dev_mask() == Context::kGPU) {
     cuda_stream = static_cast<void*>(gpu_stream->stream_);
   }
 #endif
 
+  // get mxnet initialized and seeded RNG states and pass to lib_api.h
+  void *rng_cpu_states = nullptr, *rng_gpu_states = nullptr;
+  if (!is_subgraph_op) {
```

Comment:
why are we only doing this for non-subgraph ops? Subgraph ops cant get RNG?

Reply:
subgraph op is using DefaultSubgraphOpResourceRequest, so if no operator in the subgraph resource request pass has parallel random, resource request will error out

but I think I should check initialized resource types instead of check subgraph op, I will make the change

Reply:
DefaultSubgraphOpResourceRequest calls DefaultSubgraphOpResourceRequestHelper which goes through the ops in the subgraph and sets all resources requested by any op in the subgraph. One of these could be a custom op who's defining the resource request. 

You have a good point though, we should probably just have subgraph ops define all supported resources anyway.